src/Associations.cc

```diff
@@ -129,18 +125,16 @@ void Associations::associateCatalogs(const double matchCutInArcSec,
         int matchedCount = 0;
         for (auto const &starMatch: *smList)
         {
-            const BaseStar &bs = *starMatch.s1;
-            const MeasuredStar &ms_const = dynamic_cast<const MeasuredStar &>(bs);
-            MeasuredStar &ms = const_cast<MeasuredStar &>(ms_const);
-            const BaseStar &bs2 = *starMatch.s2;
-            const FittedStar &fs_const = dynamic_cast<const FittedStar &>(bs2);
-            FittedStar &fs = const_cast<FittedStar &>(fs_const);
-            ms.SetFittedStar(&fs);
+            auto bs = starMatch.s1;
+            auto ms_const = std::dynamic_pointer_cast<const MeasuredStar>(bs);
+            auto ms = std::const_pointer_cast<MeasuredStar>(ms_const);
+            auto bs2 = starMatch.s2;
+            auto fs_const = std::dynamic_pointer_cast<const FittedStar>(bs2);
+            auto fs = std::const_pointer_cast<FittedStar>(fs_const);
+            ms->SetFittedStar(fs);
             matchedCount++;
         }
         LOGLS_INFO(_log, "Matched " << matchedCount << " objects in " << ccdImage->getName());
-        // delete the matches
-        delete smList;
```

Comment:
Because `smList` is a `std::unique_ptr<StarMatchList>` and is cleaned up automatically at the end of scope. We could shrink the scope here of course, but I don't think that is really needed. Feel free to push back if you do think we should add a block scope here and in the other cases.